sync/src/relayer/mod.rs

```diff
@@ -168,8 +168,10 @@ impl<CS: ChainStore + 'static> Relayer<CS> {
         message: RelayMessage,
     ) {
         if let Err(err) = self.try_process(Arc::clone(&nc), peer, message) {
-            debug_target!(crate::LOG_TARGET_RELAY, "try_process error {}", err);
-            nc.ban_peer(peer, BAD_MESSAGE_BAN_TIME);
+            if let Ok(Error::Misbehavior(e)) = err.downcast() {
+                debug_target!(crate::LOG_TARGET_RELAY, "try_process error {}", e);
+                nc.ban_peer(peer, BAD_MESSAGE_BAN_TIME);
+            }
```

Comment:
the original logical will ban malformed message, `Misbehavior` didn't cover this

Reply:
updated